lisa/features/gpu.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from enum import Enum
+from typing import Any, cast
+
+from lisa.base_tools.wget import Wget
+from lisa.feature import Feature
+from lisa.operating_system import Linux, Redhat, Ubuntu
+from lisa.tools import Uname
+from lisa.util import LisaException, SkippedException
+from lisa.util.logger import get_logger
+
+FEATURE_NAME_GPU = "Gpu"
+
+ComputeSDK = Enum(
+    "ComputeSDK",
+    [
+        # GRID Driver
+        "GRID",
+        # CUDA Driver
+        "CUDA",
+    ],
+)
+
+# Link to the latest GRID driver
+# The DIR link is
+# https://download.microsoft.com/download/9/5/c/95c667ff-ab95-4c56-89e0-e13e9a76782d/NVIDIA-Linux-x86_64-460.32.03-grid-azure.run
+GRID_DRIVER = "https://go.microsoft.com/fwlink/?linkid=874272"
+
+
+class Gpu(Feature):
+    def __init__(self, node: Any, platform: Any) -> None:
+        super().__init__(node, platform)
+        self._log = get_logger("gpu", self.name(), self._node.log)
+
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_GPU
+
+    def _is_supported(self) -> bool:
+        raise NotImplementedError()
+
+    # download and install NVIDIA grid driver
+    def _install_grid_driver(self, version: str) -> None:
+        self._log.debug("Starting GRID driver installation")
+        if not version.strip():
+            version = GRID_DRIVER
+
+        # download and install the NVIDIA GRID driver
+        wget_tool = self._node.tools[Wget]
+        grid_file_path = wget_tool.get(
+            f"{GRID_DRIVER}",
+            str(self._node.working_path),
+            "NVIDIA-Linux-x86_64-grid.run",
+            executable=True,
+        )
+        result = self._node.execute(
+            f"{grid_file_path} --no-nouveau-check --silent --no-cc-version-check"
+        )
+        if result.exit_code != 0:
+            raise LisaException(
+                "Failed to install the GRID driver! " f"stdout: {result.stderr}"
+            )
+        else:
+            self._log.debug("Successfully installed the GRID driver")
+
+    # download and install CUDA Driver
+    def _install_cuda_driver(self, version: str) -> None:
+        self._log.debug("Starting CUDA driver installation")
+        cuda_repo = ""
+        linux_os: Linux = cast(Linux, self._node.os)
+
+        if not version.strip():
+            version = "10.1.105-1"
+
+        os_version = self._node.os.os_version
+
+        # CUDA driver installation for redhat distros
+        if isinstance(self._node.os, Redhat):
+            release = os_version.release.split(".")[0]
+            cuda_repo_pkg = f"cuda-repo-rhel{release}-{version}.x86_64.rpm"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/"
+                f"compute/cuda/repos/rhel{release}/x86_64/{cuda_repo_pkg}"
+            )
+            linux_os = Redhat(self._node.os)
```

Comment:
This line is not necessary, should be removed.

Reply:
Actually we dnt need the linux_os as casting up there, now that we are using install_packages_from_url()
Removed.

Reply:
Whether it's used or not, it shouldn't create an new OS instance. It will cause unexpected behavior.